src/pool/srv_target.c

```diff
@@ -867,14 +867,24 @@ ds_pool_tgt_map_update(struct ds_pool *pool, struct pool_buf *buf,
 		if (map != NULL) {
 			struct pool_map *tmp = pool->sp_map;
 
+			rc = pl_map_update(pool->sp_uuid, map,
```

Comment:
At the line 890, the update_child_map() will trigger pl_map_update() finally, right?

Reply:
How?

Reply:
ds_pool_tgt_map_update() => dss_task_collective(update_child_map) => ds_rebuild_pool_map_update() => dc_pool_update_map() => pool_map_update().

Reply:
ah, this is for client stack, I would not suggest to use this.

Reply:
But your patch will take effect under the same condition check as original calling update_child_map() does, right?

Reply:
Not sure what you mean? if there are no rebuild, dc_pool_update_map() will not be called, see ds_rebuild_pool_map_update().

Reply:
But for rebuild case, the new pl_map_update() will be redundant, right?

Reply:
Again, that dc_pool_update_map is for client stack, client I/O will share that part for refreshing the placement map during pool map update.

Reply:
it seems to me that L870 is to update the server-side cached pl_map based on the pool_map.
For rebuild case, as it needs to go through client IO stack so with dc_pool handle, so in ds_rebuild_pool_map_update() -> dc_pool_update_map() update the cached pl_map and dc_pool::dp_map. So for rebuild it does do the pl_map_update() more than one time, but the 2nd only lookup hashtable and increase pl_connects if needed. 
The problem seems due to that for both server and client side we cache the pl_map on the hash-table, but server-side rebuild need to enable client IO stack. so can see the pl_map_update twice. probably can be refined a little bit later, maybe fine for now.